src/OrleansRuntime/Streams/PersistentStream/PersistentStreamPullingAgent.cs

```diff
@@ -131,27 +134,42 @@ internal class PersistentStreamPullingAgent : SystemTarget, IPersistentStreamPul
             // Setup a reader for a new receiver. 
             // Even if the receiver failed to initialise, treat it as OK and start pumping it. It's receiver responsibility to retry initialization.
             var randomTimerOffset = safeRandom.NextTimeSpan(config.GetQueueMsgsTimerPeriod);
-            timer = base.RegisterTimer(AsyncTimerCallback, QueueId, randomTimerOffset, config.GetQueueMsgsTimerPeriod);
+            timer = RegisterTimer(AsyncTimerCallback, QueueId, randomTimerOffset, config.GetQueueMsgsTimerPeriod);
 
-            logger.Info((int) ErrorCode.PersistentStreamPullingAgent_04, "Taking queue {0} under my responsibility.", QueueId.ToStringWithHashCode());
+            logger.Info((int)ErrorCode.PersistentStreamPullingAgent_04, "Taking queue {0} under my responsibility.", QueueId.ToStringWithHashCode());
         }
 
         public async Task Shutdown()
         {
             // Stop pulling from queues that are not in my range anymore.
             logger.Info(ErrorCode.PersistentStreamPullingAgent_05, "Shutdown of {0} responsible for queue: {1}", GetType().Name, QueueId.ToStringWithHashCode());
+
             if (timer != null)
             {
-                var tmp = timer;
+                IDisposable tmp = timer;
                 timer = null;
                 Utils.SafeExecute(tmp.Dispose);
             }
 
+            Task localRecieverInitTask = recieverInitTask;
+            recieverInitTask = null;
+            if (localRecieverInitTask != null)
+            { 
+                try
+                {
+                    await localRecieverInitTask;
```

Comment:
If we need a timeout for fail fast, we should do it at the top level call, not a each external call.